src/libraries/System.Private.CoreLib/src/System/IO/RandomAccess.Unix.cs

```diff
@@ -70,7 +70,11 @@ internal static unsafe long ReadScatterAtOffset(SafeFileHandle handle, IReadOnly
                 for (int i = 0; i < buffersCount; i++)
                 {
                     Memory<byte> buffer = buffers[i];
-                    MemoryHandle memoryHandle = buffer.Pin();
+                    MemoryHandle memoryHandle;
+                    unsafe
+                    {
+                        memoryHandle = buffer.Pin();
+                    }
                     vectors[i] = new Interop.Sys.IOVector { Base = (byte*)memoryHandle.Pointer, Count = (UIntPtr)buffer.Length };
```

Comment:
Please put MemoryHandle inside the unsafe block, and the following two statements as well.